iib/workers/tasks/opm_operations.py

```diff
@@ -415,6 +420,59 @@ def opm_generate_dockerfile(
     return dockerfile_path
 
 
+def insert_cache_into_dockerfile(dockerfile_path: str, local_cache_path: str) -> None:
+    """
+    Insert built cache into the Dockerfile.
+
+    :param str dockerfile_path: path to the generated Dockerfile.
+    :param str local_cache_path: path to the locally generated cache.
+    """
+    with open(dockerfile_path, 'r') as f:
+        file_data = f.read()
+
+    file_data = file_data.replace(
+        'RUN ["/bin/opm", "serve", "/configs", "--cache-dir=/tmp/cache", "--cache-only"]',
+        f'COPY {local_cache_path} /tmp/cache',
+    )
+
+    with open(dockerfile_path, 'w') as f:
+        f.write(file_data)
+
+
+def generate_cache_locally(base_dir: str, fbc_dir: str, local_cache_path: str) -> None:
+    """
+    Generate the cache for the index image locally before building it.
+
+    :param str base_dir: base directory where cache should be created.
+    :param str fbc_dir: directory containing file-based catalog (JSON or YAML files).
+    :param str local_cache_path: path to the locally generated cache.
+    :return: Returns path to generated cache
+    :rtype: str
+    :raises: IIBError when cache was not generated
+
+    """
+    from iib.workers.tasks.utils import run_cmd
+
+    cmd = [
+        'opm',
+        'serve',
+        os.path.abspath(fbc_dir),
+        f'--cache-dir={local_cache_path}',
+        '--cache-only',
```

Comment:
Note: Using this tag means that this patch depends on OPM update issue since we are using v1.23.2 in our environments and this parameter was introduced later.

Reply:
Yes, that's right. The OPM upgrade commit and this one has to go out together for this fix to work

Reply:
Good point Jan. We should ensure that both opm update and this change should go in the same release.